File: ProjectManager-WebApi/Models/Programador.cs
using System.Collections.Generic;

namespace ProjectManager_WebApi.Models
{
    public class Programador
    {
        public int Id { get; set; }
        public string Nome { get; set; }
        public string Telefone { get; set; }
        public Projeto Projeto {get; set;}
        public int ProjetoId {get;set;}


        public Programador(int Id, string Nome, string Telefone, int ProjetoId)
        {
            this.Id = Id;
            this.Nome = Nome;
            this.Telefone = Telefone;
            this.ProjetoId = ProjetoId;
        }

        public Programador() { }
    }
    
}

File: ProjectManager/src/app/models/Programador.ts
import { Projeto } from './Projeto';

export class Programador{

    constructor() {
        this.id = 0,
        this.nome = '',
        this.telefone ='',
        this.projeto = null,
        this.projetoId = 0 
    }
    
    id:Number;
    nome:String;
    telefone:String;
    projeto:Projeto;
    projetoId:Number;
}